src/mapping.ts

```diff
@@ -104,13 +104,11 @@ export function handleNewHash(event: NewHash): void {
       }
     } else if (transaction.isSet("data")) {
       let data = transaction.getEntry("data")!.value;
-        try {
-          entity.data = data.toString();
-        }
-        catch (e) {
-          log.warning("IPFS document {} has a invalid data field", [ipfsHash]);
-          return;
-        }
+      if(data.kind === JSONValueKind.STRING) {
+        log.warning("IPFS document {} has a invalid data field", [ipfsHash]);
+        return;
+      }
+      entity.data = data.toString();
```

Comment:
Shouldn't we still wrap the `try` `catch` here? What if they user other types like numbers etc?

Reply:
There's no try-catch in AssemblyScript. My assumption is that the string type is the only one to cause issues, by looking at https://github.com/graphprotocol/graph-tooling/blob/57214053e8daa129a64dda4acf1302cc81a56dc3/packages/ts/common/value.ts#L80C9-L80C9